### ISSUE TITLE
New Resource request: Network Watcher Flow Log

### ISSUE THREAD

**Issue by liemnotliam** [terraform-providers/terraform-provider-azurerm#1776]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Add support for a new azurerm resource to create and configure [network watcher NSG flow logs](https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-nsg-flow-logging-overview).

### New or Affected Resource(s)

* new: azurerm_network_watcher_flow_log

### Potential Terraform Configuration

```hcl
resource "azurerm_network_watcher_flow_log" "test {
    resource_group_name  = "test-rg"
    network_watcher_name = "eastus-networkwatcher"
    
    enabled            = true
    target_resource_id = "/subscriptions/########-####-####-####-############/resourceGroups/test-rg/providers/Microsoft.Network/networkSecurityGroups/test-nsg"
    storage_account_id = "/subscriptions/########-####-####-####-############/resourceGroups/test-rg/providers/Microsoft.Storage/storageAccounts/networkwatchertestsa"

    retention_policy {
        enabled = true
        days = 7
    }

    traffic_analytics {
        enabled               = true
        workspace_id          = "########-####-####-####-############"
        workspace_region      = "eastus"
        workspace_resource_id = "/subscriptions/########-####-####-####-############/resourcegroups/test-rg/providers/microsoft.operationalinsights/workspaces/test-workspace" 
    }
}
```

### References

* https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-04-01/network#WatchersClient.SetFlowLogConfiguration

**Comment by VikneshR** [comment]
Is there any solution for this case?

**Comment by TalalMasood** [comment]
Hi, what is the expected availability for this resource ? (azurerm_network_watcher_flow_log)

**Comment by beaudryj** [comment]
This ⏫

**Comment by mamade78** [comment]
Hello any update for this request ?

**Comment by tombuildsstuff** [comment]
@mamade78

> Hello any update for this request ?

As mentioned in the PR - unfortunately this is currently blocked on an issue in the API which is being tracked here: https://github.com/Azure/azure-sdk-for-go/issues/4781. Once the upstream issue's been resolved we'll take another look at this - but unfortunately at this time there's not a lot we can do with the API as-is.

Thanks!

**Comment by adamday2** [comment]
@tombuildsstuff although its not completely clear based on the ticket status, it looks like this might be fixed in the upstream now.

**Comment by JayDoubleu** [comment]
Is there any update on this ? It would be a great resource to use.

**Comment by draggeta** [comment]
Yeah, if this is fixed upstream, I'd love to see liemnotliam's PR merged.

**Comment by bostonmoto** [comment]
Hi @tombuildsstuff there's a note on the referenced blocker https://github.com/Azure/azure-sdk-for-go/issues/4781 saying the issue was fixed on 8/21. Is it possible to proceed with the request at this point?
Thanks!

**Comment by ymash13** [comment]
It would be really nice to have this done. I really need it.

**Comment by schlbra** [comment]
@tombuildsstuff , can you guys please take another look at this since it appears the upstream API has been fixed?

**Comment by JimmyKarlsson1** [comment]
Any news on this one?
